lisa/operating_system.py

```diff
@@ -781,7 +781,10 @@ def add_repository(
     def _initialize_package_installation(self) -> None:
         # wait running system package process.
         self.wait_running_package_process()
-
+        if type(self._node.os) == Ubuntu:
+            # wait till cloud-init finish to run the init work include updating
+            # /etc/apt/source.list file
```

Comment:
Add comments that, this applies to Azure only. It's ok for current logic, because we don't check its result.